ci/check-style.sh

```diff
@@ -25,36 +25,39 @@ fi
 
 readonly BINDIR="$(dirname "$0")"
 
-find google/cloud -name '*.h' -print0 \
-  | xargs -0 awk -f "${BINDIR}/check-include-guards.gawk"
+# Build paths to ignore in find(1) commands by reading .gitignore.
+declare -a ignore=( -path ./.git )
+if [ -f .gitignore ]; then
+  while read -r line; do
+    case "${line}" in
+    [^#]*/*) ignore+=( -o -path "./$(expr "${line}" : '\(.*\)/')" );;
+    [^#]*)   ignore+=( -o -name "${line}" );;
```

Comment:
Should this be `"./${line}"`?

Reply:
A slash in a `-name` argument will cause it to never match.  That is, `find -name` is only looking at the basename of the candidate path.

In the case of `bazel-*` from our `.gitignore`, we should probably be saying `bazel-*/'.

All that said, `find` can't implement all the `.gitignore` functionality (`**` is a simple example), so this mapping is certainly based on us only using simple "ignore" specifications.